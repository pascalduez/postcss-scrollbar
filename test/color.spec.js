import postcss from 'postcss';
import { stripIndent } from 'common-tags';
import plugin from '../src';

let from, to;
let run = input => postcss().use(plugin).process(input, { from, to });

describe('color:', () => {
  test('<color> <color>', async () => {
    let input = stripIndent`
      .test {
        scrollbar-color: rebeccapurple green;
      }
    `;

    let result = await run(input);
    expect(result.css).toMatchSnapshot();
  });

  test('keyword', async () => {
    let input = stripIndent`
      .test {
        scrollbar-color: dark;
      }
    `;

    let result = await run(input);
    expect(result.css).toMatchSnapshot();
  });

  test('functional color', async () => {
    let input = stripIndent`
      .test {
        scrollbar-color: rgb(255, 0, 0) transparent;
      }
    `;

    let result = await run(input);
    expect(result.css).toMatchSnapshot();
  });

  test('erroneous keyword', async () => {
    let input = stripIndent`
      .test {
        scrollbar-color: blue;
      }
    `;

    let result = await run(input);

    expect(result.css).toMatchSnapshot();
    expect(result.messages.length).toBeGreaterThan(0);
    expect(result.messages[0].type).toBe('warning');
    expect(result.messages[0].text).toMatch(
      /Invalid value for property `scrollbar-color`/
    );
  });
});
