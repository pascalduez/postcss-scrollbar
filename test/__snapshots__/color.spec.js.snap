// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`color:  <color> <color> 1`] = `
".test::-webkit-scrollbar-thumb {
  background-color: rebeccapurple;
}
.test::-webkit-scrollbar-track {
  background-color: green;
}
.test::-webkit-scrollbar-corner {
  background-color: green;
}
.test {
  scrollbar-color: rebeccapurple green;
}"
`;

exports[`color:  erroneous keyword 1`] = `
".test {
  scrollbar-color: blue;
}"
`;

exports[`color:  keyword 1`] = `
".test::-webkit-scrollbar-thumb {
  background-color: initial;
}
.test::-webkit-scrollbar-track {
  background-color: initial;
}
.test::-webkit-scrollbar-corner {
  background-color: initial;
}
.test {
  scrollbar-color: dark;
}"
`;
